Run rector on lib/private

CarlSchwan opened the pull request:
* Resolves: # <!-- related github issue -->

## Summary

Seems to work fine locally and the CI is also happy

I ended up doing some manual changes to lib/private/Group/Group.php, to lib/private/Server.php and to lib/private/AppFramework/DependencyInjection/DIContainer.php to make it work

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)
- [ ] [Labels added](https://github.com/nextcloud/server/labels) where applicable (ex: bug/enhancement, `3. to review`, feature component)
- [ ] [Milestone added](https://github.com/nextcloud/server/milestones) for target branch/version (ex: 32.x for `stable32`)

CarlSchwan committed:
chore(rector): Run rector on apps, core and tests directory

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor: Run rector on lib/private

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

miaulalala commented on lib/private/Console/TimestampFormatter.php:
```diff
@@ -12,18 +12,17 @@
-		$this->config = $config;
+	public function __construct(
+		protected ?IConfig $config,
+		OutputFormatterInterface $formatter,
```
```suggestion
		protected OutputFormatterInterface $formatter,
```

miaulalala commented on lib/private/DB/Connection.php:
```diff
@@ -53,7 +55,7 @@ class Connection extends PrimaryReadReplicaConnection {
 
-	/** @var \OC\DB\Adapter $adapter */
+	/** @var Adapter $adapter */
 	protected $adapter;
```
```suggestion
	protected Adapter $adapter;
```

miaulalala commented on lib/private/DB/Exceptions/DbalException.php:
```diff
@@ -35,21 +35,24 @@
 	 * @param \Doctrine\DBAL\Exception $original
 	 * @param int $code
 	 * @param string $message
 	 */
```
```suggestion
```

miaulalala commented on lib/private/DB/Migrator.php:
```diff
@@ -23,20 +23,15 @@ class Migrator {
-		IConfig $config,
-		?IEventDispatcher $dispatcher = null) {
+	public function __construct(
+		Connection $connection,
```
```suggestion
		IDBConnection $connection,
```

miaulalala commented on lib/private/DB/Migrator.php:
```diff
@@ -23,20 +23,15 @@ class Migrator {
-	private ?IEventDispatcher $dispatcher;
-
 	/** @var bool */
 	private $noEmit = false;
```
Typing

miaulalala commented on lib/private/DB/ObjectParameter.php:
```diff
@@ -11,19 +11,18 @@
-		$this->stringable = \is_callable([$object, '__toString']);
-		$this->class = \get_class($object);
+	public function __construct(
+		private $object,
```
Typing?

CarlSchwan committed:
fix(IAppContainer): Expose more correct type

OC\ServerContainer is not inherinting from IServerContainer so expose
the most near type.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

miaulalala reviewed (changes requested):
First batch

miaulalala reviewed (changes requested):
First batch

CarlSchwan committed:
refactor: Apply comments

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
fix(ExpressionBuilderTest): Fix test and improve typing

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
fix(ShareAPIControllerTest): Fix return types of DateTimeZone

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

miaulalala commented on lib/private/Files/Cache/FailedCache.php:
```diff
@@ -19,16 +19,14 @@
 	 * FailedCache constructor.
 	 *
 	 * @param bool $visible
 	 */
```
```suggestion
```

miaulalala commented on lib/private/Files/Cache/Updater.php:
```diff
@@ -27,12 +29,7 @@ class Updater implements IUpdater {
-	 * @var \OC\Files\Cache\Propagator
+	 * @var Propagator
 	 */
 	protected $propagator;
```
```suggestion
	protected Propagator $propagator;
```

miaulalala commented on lib/private/Files/Cache/Updater.php:
```diff
@@ -49,14 +46,15 @@ class Updater implements IUpdater {
 	/**
-	 * @param \OC\Files\Storage\Storage $storage
+	 * @param Storage $storage
 	 */
```
unneccessary

miaulalala commented on lib/private/Files/Config/LazyStorageMountInfo.php:
```diff
@@ -11,17 +11,17 @@
 	 *
 	 * @param IUser $user
 	 * @param IMountPoint $mount
 	 */
```
```suggestion
```

miaulalala commented on lib/private/Files/Config/UserMountCacheListener.php:
```diff
@@ -14,18 +14,14 @@
 	 * UserMountCacheListener constructor.
 	 *
 	 * @param IUserMountCache $userMountCache
 	 */
```
```suggestion
```

miaulalala commented on lib/private/Files/Mount/MountPoint.php:
```diff
@@ -52,17 +53,14 @@ class MountPoint implements IMountPoint {
 	 */
 	private $invalidStorage = false;
```
```suggestion
	private bool $invalidStorage = false;
```

miaulalala commented on lib/private/Files/Search/SearchQuery.php:
```diff
@@ -12,18 +12,6 @@
-
 	/**
 	 * SearchQuery constructor.
 	 *
```
delete

miaulalala commented on lib/private/Files/Storage/CommonTest.php:
```diff
@@ -7,15 +7,15 @@
 	/**
 	 * underlying local storage used for missing functions
-	 * @var \OC\Files\Storage\Local
+	 * @var Local
```
```suggestion
```

miaulalala commented on lib/private/Files/Storage/Home.php:
```diff
@@ -7,23 +7,28 @@
+	 * @var User $user
 	 */
 	protected $user;
 
```
typing

miaulalala commented on lib/private/Files/Storage/Wrapper/Wrapper.php:
```diff
@@ -22,9 +22,9 @@
-	 * @var \OC\Files\Storage\Storage $storage
+	 * @var Storage $storage
 	 */
 	protected $storage;
```
type

CarlSchwan replied:
This unfortunately breaks stuff. I leave it as it for now.

miaulalala commented on lib/private/Files/Template/TemplateManager.php:
```diff
@@ -41,43 +44,23 @@ class TemplateManager implements ITemplateManager {
-	private $config;
 	private $l10n;
-	private $logger;
 	private $userId;
```
```suggestion
	private ?string $userId;
```

miaulalala commented on lib/private/Files/Template/TemplateManager.php:
```diff
@@ -41,43 +44,23 @@ class TemplateManager implements ITemplateManager {
-		$this->l10n = $l10nFactory->get('lib');
-		$this->logger = $logger;
+		$this->l10n = $this->l10nFactory->get('lib');
 		$user = $userSession->getUser();
```
Injectable instead?

miaulalala commented on lib/private/Files/Utils/Scanner.php:
```diff
@@ -74,13 +65,14 @@ class Scanner extends PublicEmitter {
-		$this->logger = $logger;
+	public function __construct(
+		private $user,
+		protected $db,
```
typing?

miaulalala commented on lib/private/Files/View.php:
```diff
@@ -1515,17 +1526,17 @@ public function getDirectoryContent($directory, $mimetype_filter = '', ?\OCP\Fil
 		/**
-		 * @var \OC\Files\FileInfo[] $fileInfos
+		 * @var FileInfo[] $fileInfos
 		 */
```
inline

miaulalala commented on lib/private/Files/View.php:
```diff
@@ -1941,15 +1952,15 @@ private function getPartFileInfo(string $path): \OC\Files\FileInfo {
 		// All of the view's functions disallow '..' in the path so we can short cut if the path is invalid
 		if (!Filesystem::isValidPath($path ?: '/')) {
-			$l = \OCP\Util::getL10N('lib');
+			$l = Util::getL10N('lib');
```
declare outside of all method calls / catches and reuse?

CarlSchwan replied:
I will leave like this. View contains a  lot of spagueti code and touching one part, often breaks other parts

miaulalala commented on lib/private/GlobalScale/Config.php:
```diff
@@ -9,16 +9,14 @@
 	 * Config constructor.
 	 *
 	 * @param IConfig $config
 	 */
```
```suggestion
```

miaulalala commented on lib/private/Http/Client/Response.php:
```diff
@@ -13,11 +13,12 @@
 
-	public function __construct(ResponseInterface $response, bool $stream = false) {
+	public function __construct(
+		ResponseInterface $response,
```
injectable?

miaulalala commented on lib/private/L10N/L10N.php:
```diff
@@ -10,23 +10,12 @@
-
 	/** @var IdentityTranslator */
 	private $identityTranslator;
 
```
typing

miaulalala commented on lib/private/L10N/L10N.php:
```diff
@@ -40,12 +29,13 @@ class L10N implements IL10N {
+		protected IFactory $factory,
+		protected $app,
+		protected $lang,
+		protected $locale,
```
typing?

miaulalala commented on lib/private/L10N/L10NString.php:
```diff
@@ -8,29 +8,18 @@
 	 * @param string|string[] $text
 	 * @param array $parameters
 	 * @param int $count
 	 */
```
```suggestion
	/**
	 * @param string|string[] $text
	 */
```

miaulalala commented on lib/private/L10N/L10NString.php:
```diff
@@ -8,29 +8,18 @@
-		$this->count = $count;
+	public function __construct(
+		protected L10N $l10n,
+		protected $text,
```
could this be typed?

miaulalala commented on lib/private/L10N/LazyL10N.php:
```diff
@@ -14,12 +14,10 @@ class LazyL10N implements IL10N {
 	/** @var IL10N */
 	private $l;
```
```suggestion
	private IL10N $l;
```

miaulalala commented on lib/private/Memcache/Cache.php:
```diff
@@ -7,20 +7,18 @@
-	public function __construct($prefix = '') {
-		$this->prefix = $prefix;
+	public function __construct(
+		protected $prefix = '',
```
type?

miaulalala reviewed (changes requested):
Batch 2 :1st_place_medal:

miaulalala reviewed (changes requested):
Batch 2 :1st_place_medal:

CarlSchwan commented on lib/private/Files/Cache/LocalRootScanner.php:
```diff
@@ -8,14 +8,15 @@
-	public function __construct(\OC\Files\Storage\Storage $storage) {
+	public function __construct(Storage $storage) {
 		parent::__construct($storage);
 		$config = Server::get(IConfig::class);
```
Not without changing quite a bit of code in the unit tests and in other places in the code. Todo for later ;)

CarlSchwan commented on lib/private/Files/Utils/Scanner.php:
```diff
@@ -106,28 +98,28 @@ protected function getMounts($dir) {
+	 * @param MountPoint $mount
 	 */
 	protected function attachListener($mount) {
 		/** @var \OC\Files\Cache\Scanner $scanner */
```
The file already is called Scanner so that won't work

CarlSchwan committed:
refactor: Apply second batch of comments

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

salmart-dev commented on apps/files_external/lib/Lib/Notify/SMBNotifyHandler.php:
```diff
@@ -110,21 +110,16 @@ private function mapChange(\Icewind\SMB\Change $change) {
+	 * @return IChange::ADDED|IChange::REMOVED|IChange::MODIFIED|IChange::RENAMED|null
+	 */
+	private function mapNotifyType($smbType): ?int {
+		return match ($smbType) {
```
could import INotifyHandler to make the match more readable

salmart-dev commented on lib/private/AppFramework/Http/Request.php:
```diff
@@ -81,19 +77,17 @@ class Request implements \ArrayAccess, \Countable, IRequest {
+		protected IRequestId $requestId,
+		protected IConfig $config,
+		protected ?CsrfTokenManager $csrfTokenManager = null,
+		protected string $inputStream = 'php://input',
```
todo: check this doesn't cause issues

salmart-dev commented on lib/private/Archive/ZIP.php:
```diff
@@ -16,17 +18,13 @@ class ZIP extends Archive {
-	public function __construct(string $source) {
-		$this->path = $source;
+	public function __construct(
+		private string $path,
```
todo: check this rename can´t cause issues

salmart-dev commented on lib/private/Authentication/TwoFactorAuth/Manager.php:
```diff
@@ -37,59 +40,21 @@ class Manager {
+		private LoggerInterface $logger,
+		private TokenProvider $tokenProvider,
+		private ITimeFactory $timeFactory,
+		private IEventDispatcher $dispatcher,
```
todo: check may break because of rename

salmart-dev commented on tests/lib/Encryption/Keys/StorageTest.php:
```diff
@@ -20,7 +20,7 @@ class StorageTest extends TestCase {
 	protected $storage;
 
-	/** @var MockObject|\OC\Encryption\Util */
+	/** @var MockObject|Util */
```
```suggestion
	/** @var MockObject&Util */
```

salmart-dev reviewed (dismissed):
Besides the small issues in the comments I noticed many changes in constructor parameter names. Those would be breaking changes since PHP has named parameters support, but I assume that we don't use that internally. I think all classes I saw with this kind of change were all internal.

CarlSchwan committed:
refactor: Update repairs jobs

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

miaulalala commented on lib/private/PreviewManager.php:
```diff
@@ -249,22 +262,22 @@ protected function getEnabledDefaultProvider(): array {
+			BMP::class,
+			XBitmap::class,
+			Krita::class,
+			WebP::class,
```
I'd prefer the long class constants here

miaulalala commented on lib/private/Remote/Instance.php:
```diff
@@ -18,12 +18,6 @@ class Instance implements IInstance {
 	/** @var string */
 	private $url;
```
typing

miaulalala commented on lib/private/Remote/Instance.php:
```diff
@@ -18,12 +18,6 @@ class Instance implements IInstance {
-	private $clientService;
-
 	/** @var array|null */
 	private $status;
```
typing

miaulalala commented on lib/private/Repair/NC22/LookupServerSendCheck.php:
```diff
@@ -10,17 +10,13 @@
-		$this->config = $config;
+	public function __construct(
+		private readonly IJobList $jobList,
+	) {
```
Missing `IConfig`

CarlSchwan replied:
It's not used and the class is now exclusively autoloaded with `Server::get`

miaulalala commented on lib/private/Repair/Owncloud/UpdateLanguageCodes.php:
```diff
@@ -13,33 +13,17 @@
-	 * {@inheritdoc}
-	 */
-	public function getName() {
+	public function getName(): string {
```
Do all repair steps need the `#[\Override]` attributes for `getName()` and `run()`?

miaulalala commented on lib/private/Server.php:
```diff
@@ -259,16 +309,15 @@
 	 * @param string $webRoot
-	 * @param \OC\Config $config
+	 * @param Config $config
 	 */
```
Not needed?

miaulalala commented on lib/private/Share20/DefaultShareProvider.php:
```diff
@@ -388,9 +395,9 @@ public function getChildren(IShare $parent): array {
 	 *
-	 * @param \OCP\Share\IShare $share
+	 * @param IShare $share
 	 */
```
unnecessary

miaulalala commented on lib/private/Share20/DefaultShareProvider.php:
```diff
@@ -1400,7 +1407,7 @@ protected function filterSharesOfUser(array $shares) {
 	 * propagate notes to the recipients
 	 *
 	 * @param IShare $share
```
```suggestion
```

miaulalala commented on lib/private/Share20/DefaultShareProvider.php:
```diff
@@ -1538,7 +1545,7 @@ protected function sendUserShareMail(
 	 *
 	 * @param array $recipients
 	 * @param IShare $share
```
```suggestion
```

miaulalala commented on lib/private/Share20/Manager.php:
```diff
@@ -1388,13 +1395,13 @@ public function getShareById($id, $recipient = null, bool $onlyValid = true) {
-	 * @param \OCP\Files\Node $path
+	 * @param Node $path
 	 * @param int $page
 	 * @param int $perPage
```
typing?

miaulalala commented on lib/private/Share20/Manager.php:
```diff
@@ -1632,12 +1639,12 @@ public function userDeletedFromGroup($uid, $gid) {
-	 * @param \OCP\Files\Node $path
+	 * @param Node $path
 	 * @param bool $recursive Should we check all parent folders as well
 	 * @param bool $currentAccess Ensure the recipient has access to the file (e.g. did not unshare it)
```
typing?

miaulalala commented on lib/private/TempManager.php:
```diff
@@ -12,32 +12,31 @@
-	/** @var IConfig */
-	protected $config;
 	/** @var IniGetWrapper */
 	protected $iniGetWrapper;
```
typing? Or could this be injected?

miaulalala commented on lib/private/URLGenerator.php:
```diff
@@ -18,53 +18,40 @@
-	/** @var Router */
-	private $router;
 	/** @var null|string */
 	private $baseUrl = null;
```
typing

miaulalala commented on lib/private/legacy/OC_App.php:
```diff
@@ -547,19 +560,19 @@ public static function executeRepairSteps(string $appId, array $steps) {
-	 * @param \OCP\IL10N $l
+	 * @param IConfig $config
+	 * @param IL10N $l
 	 * @param array $info
```
:wastebasket:

miaulalala commented on lib/private/legacy/OC_User.php:
```diff
@@ -133,30 +143,31 @@ public static function setupBackends() {
 	 * Log in a user and regenerate a new session.
 	 *
-	 * @param \OCP\Authentication\IApacheBackend $backend
+	 * @param IApacheBackend $backend
```
```suggestion
```

miaulalala commented on lib/public/Files/Cache/IUpdater.php:
```diff
@@ -57,7 +53,7 @@ public function remove($path);
 	 * @param string $target
```
```suggestion
```

miaulalala commented on lib/public/IServerContainer.php:
```diff
@@ -57,7 +57,7 @@ public function getRootFolder();
 	 * Returns a view to ownCloud's files folder
```
```suggestion
	 * Returns a view to Nextcloud's files folder
```

miaulalala commented on tests/lib/Collaboration/Collaborators/MailPluginTest.php:
```diff
@@ -1021,7 +1021,7 @@ function ($appName, $key, $default) {
 		$this->instantiatePlugin(IShare::TYPE_USER);
 
-		/** @var \OCP\IUser | \PHPUnit\Framework\MockObject\MockObject */
+		/** @var IUser|\PHPUnit\Framework\MockObject\MockObject */
```
```suggestion
		/** @var IUser&\PHPUnit\Framework\MockObject\MockObject */
```

miaulalala commented on tests/lib/Collaboration/Collaborators/MailPluginTest.php:
```diff
@@ -1021,7 +1021,7 @@ function ($appName, $key, $default) {
 
-		/** @var \OCP\IUser | \PHPUnit\Framework\MockObject\MockObject */
+		/** @var IUser|\PHPUnit\Framework\MockObject\MockObject */
 		$currentUser = $this->createMock('\OCP\IUser');
```
```suggestion
		$currentUser = $this->createMock(IUser::class);
```

miaulalala commented on tests/lib/ContactsManagerTest.php:
```diff
@@ -12,7 +12,7 @@
 class ContactsManagerTest extends \Test\TestCase {
-	/** @var \OC\ContactsManager */
+	/** @var ContactsManager */
 	private $cm;
```
typing?

miaulalala reviewed (changes requested):
didn't check the rest of the test files, I assume they could also be strongly typed.

Last batch :partying_face:

CarlSchwan commented on lib/private/Repair.php:
```diff
@@ -153,94 +147,78 @@ public function addStep($repairStep) {
+			Server::get(AddCleanupDeletedUsersBackgroundJob::class),
+			Server::get(SanitizeAccountProperties::class),
+			Server::get(AddMovePreviewJob::class),
+			Server::get(ConfigKeyMigration::class),
```
yes, I checked all these class but the first one can be autoloaded. This makes it a bit easier to change the constructor later one if needed

CarlSchwan committed:
refactor: Add more typing

- repairs job
- database
- redis

And remove Helpertest which was unused outside of some tests.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan commented on lib/private/Preview/WatcherConnector.php:
```diff
@@ -23,17 +24,17 @@ public function __construct(
 
 	private function getWatcher(): Watcher {
-		return \OCP\Server::get(Watcher::class);
+		return Server::get(Watcher::class);
```
done

CarlSchwan commented on lib/private/Preview/WatcherConnector.php:
```diff
@@ -23,17 +26,21 @@ public function __construct(
 		// Do not connect if we are not setup yet!
 		if ($this->config->getValue('instanceid', null) !== null) {
-			$this->root->listen('\OC\Files', 'postWrite', function (Node $node) {
+			$this->root->listen('\OC\Files', 'postWrite', function (Node $node): void {
```
If I remember correctly with hooks/listen, the string need to match and `\OC\Files::class` is equal to `OC\Files` and not `\OC\Files`. We had this sort of issue in the past, where a refactoring broke things

miaulalala reviewed (approved):
Small nitpick, looks good!

CarlSchwan committed:
refactor(navigation-manager): Cleanup implementation and add type hinting

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

susnux reviewed (approved):
Changes look sane

CarlSchwan committed:
fix(WatcherTest): Properly mock getCache and getScanner in IStorage

Otherwise we end up with trying to assign null to a property which has
strict types.

Signed-off-by: Carl Schwan <carlschwan@kde.org>
